app_feup/lib/controller/networking/network_router.dart

```diff
@@ -37,6 +38,8 @@ class NetworkRouter {
     if (response.statusCode == 200) {
       final Session session = Session.fromLogin(response, faculties);
       session.persistentSession = persistentSession;
+      final printUpCookie = await printLogin(user, pass);
+      session.cookies += '; ' + printUpCookie;
```

Comment:
If this fails (as it does now since the service is down), the app should not block. Perhaps you could, on the print fetcher, check if the login is up, if not, if the last app login (save it in the Redux store) was older than say 7 days, try to login again in the background.